### PR TITLE
Auth documentation

### PR DESCRIPTION
**Pull request by tanay13**
## Issue that this pull request solves

 Closes: #356 

### Brief description of what is fixed or changed
Auth module documentation

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by tanay13** [comment]
@jackfrost13  please review this

File: backend/docs/admin.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+</br>
+
+### Admin module
+The Admin module is responsible for all the functions related to admin , for example : fetching particular admin data from the database , creating a new account , changing password etc. 
+Before executing every route function validation middleware gets executed which checks whether the input is in the given format or not . For this validation process a library called joi is used. 
+### API description
+- <code> <b> /admin </b> </code>
+    - method : <b>GET</b>
+    - response body : 
+        | Parameter   |
```
Comment (Kajol-Kumari) [inline backend/docs/admin.md:11]: Can we please have a single line in `response` section of each api describing its functionality for example here we can write `Get the list of all admins`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Just a minor suggestion. Rest looks nice and clean :)

File: backend/docs/admin.md (hunk 1 above)
Comment (tanay13) [reply backend/docs/admin.md:11]: @Kajol-Kumari  yes sure !!

Comment (tanay13) [reply backend/docs/admin.md:11]: @Kajol-Kumari  the part you mentioned is from admin documentation so should i remove admin doc from this branch as i have already created another PR for that . I will remove admin doc from this branch as this is just for auth doc . Is this fine?

Comment (tanay13) [reply backend/docs/admin.md:11]: @Kajol-Kumari Can you check PR #323 . That is for admin documentation . I added those lines too.

**Review by jackfrost13 — CHANGES_REQUESTED** [review]
1. It will be nice if you can add a little more detail about the JWT authentication.
2. Please share all the images with hitktechcommunity@gmail.com and transfer the ownership

**Comment by tanay13** [comment]
@jackfrost13 I made the changes please review

File: backend/docs/auth.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# <u>Backend Documentation</u>
+</br>
+
+### Auth module
+This module is responsible for all the activities related to authentication.
+
+### API description
+-  [ <b>POST</b> ] : <code> <b> /auth/login </b> </code>
+    - For login
```
Comment (Kajol-Kumari) [inline backend/docs/auth.md:9]: Please replace this line with `Login for existing users`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@tanay13 just minor changes required. Also let's follow `, ` everywhere i.e a space after comma. The doc will look cleaner that way :)

**Comment by tanay13** [comment]
@Kajol-Kumari I made the changes as suggested , please review

File: backend/docs/auth.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        | password    | string , required|
+
+    - ![diagram](images/loginroute.jpg)
+    <br>
+##### The auth middleware uses JWT method for verification.
+<br>
+
+![diagram](images/jwt.jpg)
+<br>
+JWT consists of three parts headers , payload and signature.
```
Comment (jackfrost13) [inline backend/docs/auth.md:23]: Readability can be improved over here
Please follow github markdown rules, Yours arn't working i guess.
Whenever you need to use a comma the gramattical syntax is like
`Hi, this is how we use comma` i.e after comma we leave a space and not before. Would be nice if you can refactor

File: backend/docs/auth.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+<b>Header</b> - This contains the type of token and the algorith used for signing.
+<b>Payload</b> - This contains different types of claims like registerd , private and public claims. We should avoid putting any sensitive information in the header and payload part unless it is encrypted.
```
Comment (jackfrost13) [inline backend/docs/auth.md:25]: Please explain what details we are sharing in the payload. You can learn about it using [this](https://jwt.io/)

File: backend/docs/auth.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# <u>Backend Documentation</u>
+</br>
+
+### Auth module
+This module is responsible for all the activities related to authentication.
+
+### API description
+-  [ <b>POST</b> ] : <code> <b> /auth/login </b> </code>
```
Comment (jackfrost13) [inline backend/docs/auth.md:8]: ~[ <b>POST</b> ] : <code> <b> /auth/login </b> </code>~
`/auth/login` [POST]

This looks better what do you say @Kajol-Kumari

**Review by jackfrost13 — CHANGES_REQUESTED** [review]

File: backend/docs/auth.md (hunk 5 above)
Comment (Kajol-Kumari) [reply backend/docs/auth.md:8]: Yeah. Looks cleaner!

**Comment by tanay13** [comment]
@jackfrost13 Please review , i made the changes

**Review by jackfrost13 — APPROVED** [review]

**Pull request closed without merging**